src/store.rs

```diff
@@ -5,83 +5,80 @@ use crate::error::Result;
 use crate::gc::closure;
 use crate::hash::{CidHashMap, CidHashSet, Hash};
 use crate::ipld::Ipld;
-use async_std::sync::RwLock;
-use async_trait::async_trait;
 use core::ops::Deref;
 use dag_cbor::{ReadCbor, WriteCbor};
-use futures::join;
 use std::collections::HashMap;
 use std::mem;
 use std::path::Path;
 use std::sync::Arc;
+use std::sync::RwLock;
 
 /// Implementable by ipld storage backends.
-#[async_trait]
+
```

Comment:
The Store should remain async.

Reply:
Or at least it's a completely different issue and unrelated to #19. It's also not on the hotpath and unlikely to have an large inpact on performance imo